core/src/main/scala/shapeless/generic.scala

```diff
@@ -1017,7 +1021,7 @@ class GenericMacros(val c: whitebox.Context) extends CaseClassMacros {
   }
 
   def mkProductGeneric(tpe: Type): Tree = {
-    val repr = mkHListTypTree(fieldsOf(tpe).map(_._2))
+    val repr = mkHListTpe(fieldsOf(tpe).map(_._2))
```

Comment:
Now `mkHListTypTree` is not used in shapeless anymore.
But it might be wise to keep it because people might be using it in their own macros.

Reply:
I'm fine removing it for 2.4.0.

Reply:
Ok it turns out there is a lot of dead code already 😄